File: test/golden/verify_v2.py
"""Decode a golden payload_v2.bin and verify field values."""
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parent.parent / "test_py"))
from schema_v2 import Payload  # noqa: E402


def main() -> None:
    if len(sys.argv) != 2:
        print("usage: verify_v2.py <payload_v2.bin>")
        sys.exit(1)
    data = Path(sys.argv[1]).read_bytes()
    p = Payload()
    offset = p.deserialize(data, 0)
    if offset != len(data):
        print(f"consumed {offset} of {len(data)}")
        sys.exit(1)
    if p.id != 1234567890 or p.uuid != "abc" or not p.active:
        print(f"fields mismatch: id={p.id} uuid={p.uuid} active={p.active}")
        sys.exit(1)
    if len(p.tags) != 2 or p.tags[0] != "go" or p.tags[1] != "rust":
        print(f"tags mismatch: {p.tags}")
        sys.exit(1)
    print("Python golden verify: PASSED")


if __name__ == "__main__":
    main()

File: test/test_ts/verify_v2.ts
import { Payload, BufferReader } from "./schema_v2.ts";

const path = Deno.args[0];
if (!path) {
    console.error("usage: verify_v2.ts <payload_v2.bin>");
    Deno.exit(1);
}

const data = await Deno.readFile(path);
const reader = new BufferReader(data);
const p = new Payload();
p.deserialize(reader);

if (p.id !== 1234567890n || p.uuid !== "abc" || !p.active) {
    console.error("fields mismatch", p);
    Deno.exit(1);
}
if (p.tags.length !== 2 || p.tags[0] !== "go" || p.tags[1] !== "rust") {
    console.error("tags mismatch", p.tags);
    Deno.exit(1);
}
console.log("TypeScript golden verify: PASSED");